contrib/win32/openssh/OpenSSHBuildHelper.psm1

```diff
@@ -253,15 +252,71 @@ function Start-OpenSSHBootstrap
         }
     }
 
-    #use vs2017 build tool if exists
-    if($VS2019Path -or $VS2017Path)
+    # check for corresponding build tools in the following vs order: 2019, 2017, 2015
+    # environment variable is set upon install up until vs2015 but not for newer versions
+    if($VS2019Path)
     {
-        if(-not (Test-Path $VcVars))
+        if ($null -eq $env:VS160COMNTOOLS)
```

Comment:
Please move this to a common function for both vs2019 , vs2017.

get_build_tools_path($VSInstallPath)
{
...
return build_tools_path or $null
}